### ISSUE TITLE
upgrade to k8s 1.25

### ISSUE THREAD

**Issue by parmentelat** [sopnode/kubernetes#9]
plan is to wait for fedora-37 (announced oct 18) that comes with k8s 1-25 (1.25.2-1 as of now in base and testing)

as per Damien, this might imply an upgrade of the tigera-operator pinned version as well, to be confirmed
(but there is no requirement on our end about having a newer calico)

**Comment by dsaucez** [comment]
Would it be possible to upgrade to tigera 3.24.1 as well so we are all on the same page.

Ansible provisioning inspired by your script is available on https://github.com/dsaucez/SLICES/blob/26-port-ansible-to-sonic/sopnode/ansible/roles/ here some excerpts that could be useful for you (still ongoing work to be more flexible)

For tigera:

```yaml
- name: Create tigera-operator namespace 
  kubernetes.core.k8s:
    name: tigera-operator
    api_version: v1
    kind: Namespace
    state: present

- name: Set Calico installation instructions
  ansible.builtin.template:
    src: tigera_operator.yaml.j2
    dest: tigera_operator.yaml

- name: Register project-calico helm repo
  shell: 'helm repo add projectcalico https://projectcalico.docs.tigera.io/charts'

- name: Install Calico
  shell: 'helm install calico projectcalico/tigera-operator --version v3.24.1 -f tigera_operator.yaml --namespace tigera-operator'
```

`tigera_operator.yaml`:
```yaml 
installation:
  cni:
    type: Calico
  calicoNetwork:
    bgp: Disabled
    nodeAddressAutodetectionV4:
      {{ k8s.calico.nodeAddressAutodetectionV4 | to_yaml}}
    ipPools:
    - cidr: '{{ k8s.podSubnet | default('10.244.0.0/16') }}'
      encapsulation: {{ k8s.calico.encapsulation | default('VXLAN') }}
```

I also use jinja2 yaml templates for `kubeadm` `init` and `join` operations (note that I use `cri-dockerd` should work the same with `crio`, not tested yet, hence the hardcoded socket path):

```yaml
- name: Initialize the master node
  shell: 'kubeadm init --upload-certs --config kubeadm_config.yaml'
  register: kubeadm_init
```

```yaml
apiVersion: kubeadm.k8s.io/v1beta3
kind: InitConfiguration
nodeRegistration:
  name: "{{ inventory_hostname_short }}"
  criSocket: "unix:///var/run/cri-dockerd.sock"
  kubeletExtraArgs:
    enable-controller-attach-detach: "false"
---
apiVersion: kubeadm.k8s.io/v1beta3
kind: ClusterConfiguration
networking:
  serviceSubnet: "{{ k8s.serviceSubnet | default('10.96.0.0/16') }}"
  podSubnet: "{{ k8s.podSubnet | default('10.244.0.0/16') }}"
  dnsDomain: "{{ k8s.dnsDomain | default('cluster.local') }}"
kubernetesVersion: "v1.25.2"
---
apiVersion: kubelet.config.k8s.io/v1beta1
kind: KubeletConfiguration
# kubelet specific options here
kind: KubeletConfiguration
apiVersion: kubelet.config.k8s.io/v1beta1
cgroupDriver: systemd
```

```yaml
- name: Join k8s cluster
  shell: 'kubeadm join --config ./kubeadm_config.yaml'
```

```yaml
apiVersion: kubeadm.k8s.io/v1beta3
kind: JoinConfiguration
nodeRegistration:
  name: "{{ inventory_hostname_short }}"
  criSocket: "unix:///var/run/cri-dockerd.sock"
  kubeletExtraArgs:
    enable-controller-attach-detach: "false"
discovery:
  bootstrapToken:
    apiServerEndpoint: "{{ master }}:6443"
    token: '{{ token }}'
    caCertHashes:
    - '{{ ca_cert_hash }}'
```

`ca_cert_hash` is obtained with `openssl x509 -pubkey -in /etc/kubernetes/pki/ca.crt | openssl rsa -pubin -outform der 2>/dev/null | openssl dgst -sha256 -hex | sed 's/^.* /sha256:/'` (Credit: inspired from https://medium.com/@kosta709/kubernetes-by-kubeadm-config-yamls-94e2ee11244)

and `token` with `kubeadm token create --ttl 15m`

**Comment by parmentelat** [comment]
thanks for all the details :)

just clarifying, but are you sure about the tigera version number ? 
3.24.1 feels more like a calico version number, no ?
fyi the tigera version I have pinned right now is 1.28.1...

(and yes it's confusing that tigera, whose only purpose is to install calico, has a different versioning scheme than calico; seems to have to do with the fact that tigera also supports other targets like openstack, iiuc)

**Comment by parmentelat** [comment]
Damien wrote

> Would it be possible to upgrade to tigera 3.24.1 as well so we are all on the same page.

so FYI in the f37 branch, I currently have pinned these versions

    # default for all
    export K8S_VERSION=1.25.4
    export CRIO_VERSION=1.24
    export CALICOCTL_VERSION=3.24.5
    export TIGERA_VERSION=1.28.5

let me know if further changes are needed...

**Comment by dsaucez** [comment]
As of now, I am using
```
k8s v1.25.2
CRI-O v1.24
calico v3.24.1
```

I don't care about Tigera much as it is installed as a dependency of the calico chart (projectcalico/tigera-operator --version v3.24.1 ). If needed I could fix it I guess.

I don't fix docker version but I use commit `d8accf791cbb9762a218f47507378a7c4cb63797` for cri-dockerd. Actually, I have to be a bit loose in the version for docker as I am supporting Debian 9,10,11; Fedora 36,37; and Ubuntu 20.04 that do not specially offer the same exact version.

**Comment by parmentelat** [comment]
done and tested as of v0.25
